### ISSUE TITLE
PATCH method does not parse request body

### ISSUE THREAD

**Issue by ismagilov** [svenkubiak/mangooio#298]
Controller method:
```
public Response modify(Request r, long id, TodoPatch tp) {
        ...
}
```

Model object:
```
public class TodoPatch {
    public String title;
    public Boolean completed;
    public Integer order;
}
```

Routes:
```
- method: PATCH
  url: /{id}
  mapping: ApplicationController.modify
```

PATCH request sent to http://localhost:9090/1:
```
{"title": "new title"}
```

Exception:
```
19.04.2017 13:44:14,171 [XNIO-1 I/O-6] ERROR io.mangoo.routing.handlers.ExceptionHandler - Internal server exception
java.lang.NullPointerException: null
	at io.advantageous.boon.json.implementation.JsonParserLax.parse(JsonParserLax.java:284) ~[boon-json-0.6.6.jar:?]
	at io.advantageous.boon.json.implementation.BaseJsonParser.parse(BaseJsonParser.java:126) ~[boon-json-0.6.6.jar:?]
	at io.advantageous.boon.json.implementation.BaseJsonParserAndMapper.parse(BaseJsonParserAndMapper.java:766) ~[boon-json-0.6.6.jar:?]
	at io.advantageous.boon.json.implementation.BaseJsonParserAndMapper.parse(BaseJsonParserAndMapper.java:195) ~[boon-json-0.6.6.jar:?]
	at io.advantageous.boon.json.JsonMappingParser.parse(JsonMappingParser.java:201) ~[boon-json-0.6.6.jar:?]
	at io.advantageous.boon.json.implementation.ObjectMapperImpl.readValue(ObjectMapperImpl.java:68) ~[boon-json-0.6.6.jar:?]
	at io.advantageous.boon.json.implementation.ObjectMapperImpl.fromJson(ObjectMapperImpl.java:279) ~[boon-json-0.6.6.jar:?]
	at io.mangoo.utils.JsonUtils.fromJson(JsonUtils.java:79) ~[mangooio-core-4.3.3-SNAPSHOT.jar:?]
	at io.mangoo.routing.handlers.RequestHandler.getConvertedParameters(RequestHandler.java:237) ~[mangooio-core-4.3.3-SNAPSHOT.jar:?]
	at io.mangoo.routing.handlers.RequestHandler.invokeController(RequestHandler.java:133) ~[mangooio-core-4.3.3-SNAPSHOT.jar:?]
	at io.mangoo.routing.handlers.RequestHandler.getResponse(RequestHandler.java:111) ~[mangooio-core-4.3.3-SNAPSHOT.jar:?]
	at io.mangoo.routing.handlers.RequestHandler.handleRequest(RequestHandler.java:46) ~[mangooio-core-4.3.3-SNAPSHOT.jar:?]
	at io.mangoo.routing.handlers.FormHandler.nextHandler(FormHandler.java:95) ~[mangooio-core-4.3.3-SNAPSHOT.jar:?]
	at io.mangoo.routing.handlers.FormHandler.handleRequest(FormHandler.java:37) ~[mangooio-core-4.3.3-SNAPSHOT.jar:?]
	at io.mangoo.routing.handlers.InboundCookiesHandler.nextHandler(InboundCookiesHandler.java:165) ~[mangooio-core-4.3.3-SNAPSHOT.jar:?]
	at io.mangoo.routing.handlers.InboundCookiesHandler.handleRequest(InboundCookiesHandler.java:62) ~[mangooio-core-4.3.3-SNAPSHOT.jar:?]
	at io.mangoo.routing.handlers.LocaleHandler.nextHandler(LocaleHandler.java:69) ~[mangooio-core-4.3.3-SNAPSHOT.jar:?]
	at io.mangoo.routing.handlers.LocaleHandler.handleRequest(LocaleHandler.java:58) ~[mangooio-core-4.3.3-SNAPSHOT.jar:?]
	at io.mangoo.routing.handlers.LimitHandler.nextHandler(LimitHandler.java:95) ~[mangooio-core-4.3.3-SNAPSHOT.jar:?]
	at io.mangoo.routing.handlers.LimitHandler.handleRequest(LimitHandler.java:47) ~[mangooio-core-4.3.3-SNAPSHOT.jar:?]
	at io.mangoo.routing.handlers.DispatcherHandler.nextHandler(DispatcherHandler.java:199) ~[mangooio-core-4.3.3-SNAPSHOT.jar:?]
	at io.mangoo.routing.handlers.DispatcherHandler.handleRequest(DispatcherHandler.java:171) ~[mangooio-core-4.3.3-SNAPSHOT.jar:?]
	at io.undertow.server.RoutingHandler.handleRequest(RoutingHandler.java:93) ~[undertow-core-1.4.12.Final.jar:1.4.12.Final]
	at io.undertow.server.handlers.PathHandler.handleRequest(PathHandler.java:94) ~[undertow-core-1.4.12.Final.jar:1.4.12.Final]
	at io.undertow.server.handlers.ExceptionHandler.handleRequest(ExceptionHandler.java:29) [undertow-core-1.4.12.Final.jar:1.4.12.Final]
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:211) [undertow-core-1.4.12.Final.jar:1.4.12.Final]
	at io.undertow.server.protocol.http.HttpReadListener.handleEventWithNoRunningRequest(HttpReadListener.java:244) [undertow-core-1.4.12.Final.jar:1.4.12.Final]
	at io.undertow.server.protocol.http.HttpReadListener.handleEvent(HttpReadListener.java:135) [undertow-core-1.4.12.Final.jar:1.4.12.Final]
	at io.undertow.server.protocol.http.HttpOpenListener.handleEvent(HttpOpenListener.java:148) [undertow-core-1.4.12.Final.jar:1.4.12.Final]
	at io.undertow.server.protocol.http.HttpOpenListener.handleEvent(HttpOpenListener.java:92) [undertow-core-1.4.12.Final.jar:1.4.12.Final]
	at io.undertow.server.protocol.http.HttpOpenListener.handleEvent(HttpOpenListener.java:51) [undertow-core-1.4.12.Final.jar:1.4.12.Final]
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92) [xnio-api-3.3.6.Final.jar:3.3.6.Final]
	at org.xnio.ChannelListeners$10.handleEvent(ChannelListeners.java:291) [xnio-api-3.3.6.Final.jar:3.3.6.Final]
	at org.xnio.ChannelListeners$10.handleEvent(ChannelListeners.java:286) [xnio-api-3.3.6.Final.jar:3.3.6.Final]
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92) [xnio-api-3.3.6.Final.jar:3.3.6.Final]
	at org.xnio.nio.QueuedNioTcpServer$1.run(QueuedNioTcpServer.java:128) [xnio-nio-3.3.6.Final.jar:3.3.6.Final]
	at org.xnio.nio.WorkerThread.safeRun(WorkerThread.java:580) [xnio-nio-3.3.6.Final.jar:3.3.6.Final]
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:464) [xnio-nio-3.3.6.Final.jar:3.3.6.Final]
```

It is reasonable to note, that as soon as I change in routes `PATCH` to `POST`, everything works.

**Comment by svenkubiak** [comment]
Are you sending 
`
application/json`

in the request header?

**Comment by ismagilov** [comment]
Yes, it is sent in both of cases: in POST and PATCH request: `Content-Type: application/json`.

**Comment by svenkubiak** [comment]
Fixed in 4.4.0 branch

**Comment by svenkubiak** [comment]
Please check first Beta of 4.4.0 (should be on maven central soon) and re-open if not fixed. Thanks.

**Comment by ismagilov** [comment]
Many thanks for your quick fixes! I tested the issue on `feature/4.4.0` (beta hadn't been available in maven central). It worked correctly!
